packages/block-library/src/post-date/edit.js

```diff
@@ -124,10 +119,11 @@ export default function PostDateEdit( {
 									<PublishDateTimePicker
 										currentDate={ date }
 										onChange={ setDate }
-										is12Hour={ is12HourFormat(
-											siteTimeFormat
-										) }
```

Comment:
I believe this prop is no longer needed as the proper time format for the locale is now applied via `dataOrder` prop.

Reply:
Ah, I missed that 😅 Restored the prop.